autoextract/aio/client.py

```diff
@@ -127,6 +238,15 @@ async def request():
 
     try:
         result = await request()
+    except RetryError as exc:
+        if isinstance(exc.args[0].exception(), QueryError):
```

Comment:
It seems this `try` knows a lot on how `request` behaves.
Maybe a specific exception `RetryQueryError` which inherits from `RetryError`, but is more specific and gives more context on what happened?

Reply:
I've refactored the retry logic to raise a custom/more specific exception when the retry stops because of a `QueryError`. Could you take a look and see if you like it?